File: frontend/School_Mangement/client/types/admin/student.ts
export type Student = {
    id: number;
    admissionNo: string;
    rollNo: number;
    name: string;
    gender: "Male" | "Female";
    className: string;
    section: string;
    phone: string;
    email: string;
    dateOfBirth: string;
    status: "Active" | "Inactive";
    address?: string;
};

export interface StudentFormData {
    firstName: string;
    lastName: string;
    gender: string;
    dateOfBirth: string;
    email: string;
    phone: string;
    address: string;
    schoolClass: string;
    section: string;
}

export interface StudentStats {
    totalStudents: number;
    boys: number;
    girls: number;
    newStudent: number;
}

File: backend/School-Management/src/main/java/com/School/Management/Entity/Student.java
package com.School.Management.Entity;


import com.School.Management.Enum.Status;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDate;

@Entity
@Table(name = "students")
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Student {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String admissionNo;
    private String firstName;
    private String lastName;

    private String gender;
    private LocalDate dateOfBirth;

    private String email;
    private String phone;

    private String address;

    private Integer rollNumber;

    @Enumerated(EnumType.STRING)
    private Status status;

    private LocalDate admissionDate;

    @ManyToOne
    @JoinColumn(name = "class_id")
    private SchoolClass schoolClass;

    @ManyToOne
    @JoinColumn(name = "section_id")
    private Section section;


    @OneToOne
    @JoinColumn(name = "user_id")
    private User user;

}
